src/components/DatePicker/DatePicker.tsx

```diff
@@ -47,22 +47,27 @@ export default function DatePicker(props: Props): JSX.Element {
     </>
   );
 
+  // set default timezone to UTC
+  moment.tz.setDefault('UTC');
+
   return (
     <div className={`date-picker ${props.className} ${props.errorText ? 'date-picker--error' : ''}`}>
-      <LocalizationProvider dateAdapter={AdapterDateFns}>
+      <LocalizationProvider dateAdapter={AdapterMoment} dateLibInstance={moment}>
         {props.label && (
           <label htmlFor={props.id} className='textfield-label'>
             {props.label}
           </label>
         )}
         <DesktopDatePicker
           onError={props.onError}
-          minDate={props.minDate}
-          maxDate={props.maxDate}
-          inputFormat='yyyy-MM-dd'
+          minDate={props.minDate ? moment(props.minDate) : undefined}
+          maxDate={props.maxDate ? moment(props.maxDate) : undefined}
+          inputFormat='yyyy-MM-DD'
           value={props.value ? moment(props.value, 'YYYY-MM-DD').toDate() : null}
-          onChange={(newValue: string | null) => {
-            props.onChange(props.id, newValue);
+          onChange={(newValue: any) => {
+            if (newValue?.isValid()) {
```

Comment:
Should we show an error if is not valid?

Reply:
No the user may be typing a value and it may be incomplete.

Reply:
Let me check something.

Reply:
https://momentjscom.readthedocs.io/en/latest/moment/01-parsing/15-is-valid/ checks if the date is valid
If user types '2022-' that is not valid